waspc/data/Generator/templates/server/src/_types/index.ts

```diff
@@ -24,7 +24,7 @@ export type AuthenticatedAction<Entities extends _Entity[], Input, Output> =
 type AuthenticatedOperation<Entities extends _Entity[], Input, Output> = (
   args: Input,
   context: ContextWithUser<Entities>,
-) => Promise<Output>
+) => Output | Promise<Output>
```

Comment:
We want to allow users to write Queries and Actions that aren't async (if they don't need to be).